### PR TITLE
Add SambaNova Provider

### PR DESCRIPTION
**Pull request by snova-zoltanc**

### CODE REVIEW

**Comment by snova-zoltanc** [comment]
I created a PR to add SambaNova as a provider, would appreciate help merging this! @rohitprasad15 @ksolo  @standsleeping  @andrewyng

File: aisuite/providers/sambanova_provider.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        # Ensure API key is provided either in config or via environment variable
+        config.setdefault("api_key", os.getenv("SAMBANOVA_API_KEY"))
+        if not config["api_key"]:
+            raise ValueError(
+                "Sambanova API key is missing. Please provide it in the config or set the SAMBANOVA_API_KEY environment variable."
+            )
+
+        # NOTE: We could choose to remove above lines for api_key since OpenAI will automatically
+        # infer certain values from the environment variables.
+        # Eg: OPENAI_API_KEY, OPENAI_ORG_ID, OPENAI_PROJECT_ID, OPENAI_BASE_URL, etc.
```
Comment (ksolo) [inline aisuite/providers/sambanova_provider.py:21]: This won't apply in this case, can you remove the comments?

Comment (snova-zoltanc) [reply aisuite/providers/sambanova_provider.py]: @ksolo Thanks for review, I removed these comments.

File: aisuite/providers/sambanova_provider.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        Pass the entire configuration dictionary to the OpenAI client constructor.
+        """
+        # Ensure API key is provided either in config or via environment variable
+        config.setdefault("api_key", os.getenv("SAMBANOVA_API_KEY"))
+        if not config["api_key"]:
+            raise ValueError(
+                "Sambanova API key is missing. Please provide it in the config or set the SAMBANOVA_API_KEY environment variable."
+            )
+
+        # Pass the entire config to the OpenAI client constructor
```
Comment (rohitprasad15) [inline aisuite/providers/sambanova_provider.py:19]: You wrote in the comment, but I don't think you are passing the entire config.
Assuming that is what you want, you could add/replace the base_url in config and pass the entire thing to OpenAI client.

**Review by rohitprasad15 — COMMENTED** [review]
Looks good, except for the comment.
Please test it, and also add a block in examples/client.ipynb with sambanova provider and model.

File: aisuite/providers/sambanova_provider.py (hunk 2 above)
Comment (rohitprasad15) [reply aisuite/providers/sambanova_provider.py:19]: @snova-zoltanc  - please address above. We can merge it after that. Thank you.

Comment (snova-zoltanc) [reply aisuite/providers/sambanova_provider.py:20]: @rohitprasad15 Thanks for the review I made the update

**Comment by snova-zoltanc** [comment]
@ksolo @rohitprasad15 I have addressed review, let me know if I can help with anything else before merging! 🙇

**Comment by snova-zoltanc** [comment]
@ksolo  @rohitprasad15 I have addressed review, let me know if I can help with anything else before merging! 🙇

**Review by ksolo — APPROVED** [review]

**Comment by rohitprasad15** [comment]
@snova-zoltanc  - we will just test it today on our end to see that it works, and then we can merge. Thanks for the PR.

**Comment by snova-zoltanc** [comment]
> today on our end to see that it works, and then we can merge. Thanks for the PR.

Sounds great, thank you!

**Comment by rohitprasad15** [comment]
Tested. It works. Thanks.
Screenshot 2024-12-05 at 10 06 31 AM

**Review by rohitprasad15 — APPROVED** [review]

**Pull request merged**